as-developers/rd.md

```diff
@@ -0,0 +1,175 @@
+the system should continually test at what times users are most receptive to notifications about
+upcoming deadlines, and adjust email notifications to be sent at times that users open them most
+reliably.
+
+#### 3.4.2 Functional Requirements
+
+REQ-4-1: The user's homepage should have a list of links generated based on what they visit both
+frequently and recently, based on some reasonable implementation of the "frecency" algorithm.
+
+REQ-4-2: Users should be reminded of upcoming deadlines by email, and the system should keep track
+of what email notification times are most likely to correlate with system interaction shortly after
+delivery. Then, notification delivery times should be tuned closer to these times.
+
+## 4 External Interface Requirements
+
+### 4.1 User Interfaces
+
+The user interface is the primary interface of this system, as it is otherwise quite self-contained.
+Students and instructors access the system by way of a web-app, which shows (in the case of students)
+all courses that they are registered to and can view, and (in the case of instructors) all courses
+that they administer. Users can interact with this in a standard web-app fashion (point with mouse
+pointer and click, keyboard, etc), as a later-stage goal this interface should be accessible to users
+with a wide variety of ability. In particular, it should play nice with screen readers, and shouldn't
+communicate essential information using colour in a way that would cause difficulties for people with
+colour blindness.
+
+### 4.2 Hardware Interfaces
+
+Given that this is primarily a software system, hardware details will be left to a cloud service vendor
+such as Amazon AWS, Google Cloud, or Microsoft Azure. We only require that the service be performant (as
+described in section 5 of this document), which does involve hardware considerations, but those are outside
+the scope of this document.
+
+### 4.3 Software Interfaces
+
+The system will interface with a database to durably store information.
+
+### 4.4 Communications Interfaces
+
+The system will be accessible through the Internet, so that students and instructors can view their
+courses from anywhere with an Internet connection.
+
+## 5 Other Non-Functional Requirements
+
+### 5.1 Performance Requirements
+
+The system needs to run at high speeds, 24/7, with low failure rate. Due to students needing a reliable system, it should be stress tested for 160+ hours a week and a capacity of 2500+ students at one time. System updates must happen live or nearly live as this ensures students will not need to worry about downtimes interrupting their studies.
+
+### 5.2 Security Requirements
+
+Account and data security are important as the system will be holding data of students such as names, student number, email and grades. The system will also be holding the class lectures and notes/assignments created by professors at the school which needs to protected.
```

Comment:
Some rewording for clarity

```suggestion
Account and data security are important as the system will store students' personal and private information (e.g. their names, student numbers, email addresses, and grades). The system will also store copyrighted and otherwise private information created by professors (notes, lectures, and assignments), and students (mainly assignment submissions).
```